core/includes/theme.inc

```diff
@@ -1951,7 +1951,19 @@ function theme_dropbutton_wrapper($variables) {
  */
 function theme_image($variables) {
   $attributes = $variables['attributes'];
-  $attributes['src'] = file_create_url($variables['uri']);
+
+  // Check for missing 'uri', allow D7's 'path' instead.
+  $uri = $variables['uri'];
+  if (!$uri && !empty($variables['path'])) {
+    watchdog('Image',
```

Comment:
@jlfranklin do bare in mind that my reviews are very "superficial" ...as in they come from a person still learning (a nice way to say I'm noob) and they also come in the form of questions (this helps me learn why things are done the way they are). To the point at hand...

Fair enough about the case of someone using both `'uri'` and `'path'`. As I said, my alternative code did seem to needlessly over-complicate things.

With the wrapping thing, it's just that by watching older commits and reviews made by others (that's another way I learn), I never saw function parameters being wrapped that way; only arrays. I then `grep`ed the codebase for other occurrences of `watchdog`, and didn't find any that was wrapping its parameters (even if they spanned way beyond the 80 character limit`*`). Hence my request for change in the code review.

Fair enough also about the use of `array()` vs. `NULL`. I `grep`ed wrong as it seems that there is about the same amount of occurences in the current codebase that use `NULL` as those that use `array()`. In fact, I believe that using an empty array makes it easier to add things in the future if the message is altered to contain variables. 

`*` grep `watchdog\(.{250}`:
https://github.com/backdrop/backdrop/blob/1.x/core/includes/common.inc#L783
https://github.com/backdrop/backdrop/blob/1.x/core/includes/file.inc#L795
https://github.com/backdrop/backdrop/blob/1.x/core/includes/file.inc#L1043
https://github.com/backdrop/backdrop/blob/1.x/core/modules/file/file.module#L734
https://github.com/backdrop/backdrop/blob/1.x/core/modules/image/image.effects.inc#L81
https://github.com/backdrop/backdrop/blob/1.x/core/modules/image/image.effects.inc#L132
https://github.com/backdrop/backdrop/blob/1.x/core/modules/image/image.effects.inc#L186
https://github.com/backdrop/backdrop/blob/1.x/core/modules/image/image.effects.inc#L208
https://github.com/backdrop/backdrop/blob/1.x/core/modules/image/image.effects.inc#L227
https://github.com/backdrop/backdrop/blob/1.x/core/modules/image/image.effects.inc#L274
https://github.com/backdrop/backdrop/blob/1.x/core/modules/node/node.module#L2874